src/code/z_en_item00.c

```diff
@@ -997,29 +996,72 @@ EnItem00* Item_DropCollectible2(PlayState* play, Vec3f* spawnPos, s16 params) {
-                                                  spawnPos->z, 0, 0, 0, params | param8000 | param3F00);
-            if ((spawnedActor != NULL) && !param8000) {
-                spawnedActor->actor.velocity.y = 0.0f;
-                spawnedActor->actor.speedXZ = 0.0f;
-                spawnedActor->actor.gravity = param4000 ? 0.0f : -0.9f;
-                spawnedActor->actor.world.rot.y = Rand_CenteredFloat(65536.0f);
-                spawnedActor->actor.flags |= ACTOR_FLAG_4;
-            }
+        params = Item_FlexDrop(play, spawnPos, params);
+    } 
+
+    params = func_8001F404(params & 0x00FF);
+    if (params != -1) {
+        spawnedActor = (EnItem00*)Actor_Spawn(&play->actorCtx, play, ACTOR_EN_ITEM00, spawnPos->x, spawnPos->y,
+                                                spawnPos->z, 0, 0, 0, params | param8000 | param3F00);
+        if ((spawnedActor != NULL) && !param8000) {
+            spawnedActor->actor.velocity.y = 0.0f;
+            spawnedActor->actor.speedXZ = 0.0f;
+            spawnedActor->actor.gravity = param4000 ? 0.0f : -0.9f;
+            spawnedActor->actor.world.rot.y = Rand_CenteredFloat(65536.0f);
+            spawnedActor->actor.flags |= ACTOR_FLAG_4;
         }
     }
 
     return spawnedActor;
 }
 
+u8 Item_FlexDrop(PlayState* play, Vec3f* spawnPos, s16 params) {
+    s16 dropTableIndex = Rand_ZeroOne() * 16.0f;
+    u8 dropId;
+
+    if (gSaveContext.health <= 0x10) { // 1 heart or less
+        Actor_Spawn(&play->actorCtx, play, ACTOR_EN_ELF, spawnPos->x, spawnPos->y + 40.0f, spawnPos->z, 0, 0, 0,
+                    FAIRY_HEAL_TIMED);
+        EffectSsDeadSound_SpawnStationary(play, spawnPos, NA_SE_EV_BUTTERFRY_TO_FAIRY, true,
+                                            DEADSOUND_REPEAT_MODE_OFF, 40);
+        return ITEM_NONE;
+    } else if (gSaveContext.health <= 0x30) { // 3 hearts or less
+        params = 0xB * 0x10;
+        dropTableIndex = 0x0;
+        dropId = ITEM00_RECOVERY_HEART;
+    } else if (gSaveContext.health <= 0x50) { // 5 hearts or less
+        params = 0xA * 0x10;
+        dropTableIndex = 0x0;
+        dropId = ITEM00_RECOVERY_HEART;
+    } else if ((gSaveContext.magicLevel != 0) && (gSaveContext.magic == 0)) { // Empty magic meter
+        params = 0xA * 0x10;
+        dropTableIndex = 0x0;
+        dropId = ITEM00_MAGIC_LARGE;
+    } else if ((gSaveContext.magicLevel != 0) && (gSaveContext.magic <= (gSaveContext.magicLevel >> 1))) {
+        params = 0xA * 0x10;
+        dropTableIndex = 0x0;
+        dropId = ITEM00_MAGIC_SMALL;
+    } else if (!LINK_IS_ADULT && (AMMO(ITEM_SLINGSHOT) < 6)) {
+        params = 0xA * 0x10;
+        dropTableIndex = 0x0;
+        dropId = ITEM00_SEEDS;
+    } else if (LINK_IS_ADULT && (INV_CONTENT(SLOT_BOW) != ITEM_NONE && AMMO(ITEM_BOW) < 6)) {
+        params = 0xA * 0x10;
+        dropTableIndex = 0x0;
+        dropId = ITEM00_ARROWS_MEDIUM;
+    } else if (AMMO(ITEM_BOMB) < 6) {
```

Comment:
```suggestion
    } else if (LINK_IS_CHILD && (INV_CONTENT(ITEM_SLINGSHOT) != ITEM_NONE) && (AMMO(ITEM_SLINGSHOT) < 6)) {
        params = 0xA * 0x10;
        dropTableIndex = 0x0;
        dropId = ITEM00_SEEDS;
    } else if (LINK_IS_ADULT && (INV_CONTENT(ITEM_BOW) != ITEM_NONE)&& (AMMO(ITEM_BOW) < 6)) {
        params = 0xA * 0x10;
        dropTableIndex = 0x0;
        dropId = ITEM00_ARROWS_MEDIUM;
    } else if ((INV_CONTENT(ITEM_BOMB) != ITEM_NONE) && (AMMO(ITEM_BOMB) < 6)) {
``` I think it'd be better this way